### PR TITLE
Add Verilog macro helpers to reduce duplication

### PR DESCRIPTION
**Pull request by abejgonzalez**
Use Verilog macros to instantiate wires/ports/etc in the Verilog sections of FireSim.

TL;DR: Writing Verilog without bundle connects is hard. Instead, I made AXI macros for it! Now Verilog is easy!

### Contributor Checklist
- [ ] Is this PR's title suitable for inclusion in the changelog and have you added a `changelog:<topic>` label?
- [ ] Did you add Scaladoc/docstring/doxygen to every public function/method?
- [ ] Did you add at least one test demonstrating the PR?
- [ ] Did you delete any extraneous prints/debugging code?
- [ ] Did you state the UI / API impact?
- [ ] Did you specify the Verilog / AGFI compatibility impact?

- [ ] If applicable, did you regenerate and publicly share default AGFIs?

- [ ] If applicable, did you apply the `ci:fpga-deploy` label?

- [ ] If applicable, did you apply the `Please Backport` label?

### Reviewer Checklist (only modified by reviewer)
Note: to run CI on PRs from forks, comment `@Mergifyio copy main` and manage the change from the new PR.
- [ ] Is the title suitable for inclusion in the changelog and does the PR have a `changelog:<topic>` label?
- [ ] Did you mark the proper release milestone?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?

### CODE REVIEW

File: platforms/xilinx_alveo_u250/cl_firesim/design/overall_fpga_top.v [hunk 1]
```diff
@@ -1,739 +1,196 @@
...
-    wire PCIE_M_AXI_wready;
-    wire [63:0]PCIE_M_AXI_wstrb;
-    wire PCIE_M_AXI_wvalid;
+    wire sys_reset_n;
+
+    `define AMBA_AXI4
+    `define AMBA_AXI_CACHE
+    `define AMBA_AXI_PROT
+    `define AMBA_AXI_ID
+    `AMBA_AXI_SLAVE_PORT(PCIE_M_AXI, 4, 64, 512)
```
Comment (abejgonzalez) [inline platforms/xilinx_alveo_u250/cl_firesim/design/overall_fpga_top.v:34]:
```suggestion
    `AMBA_AXI_WIRE(PCIE_M_AXI, 4, 64, 512)
```

File: platforms/xilinx_alveo_u250/cl_firesim/design/overall_fpga_top.v [hunk 2]
```diff
@@ -1,739 +1,196 @@
...
+    `undef AMBA_AXI4
+    `undef AMBA_AXI_CACHE
+    `undef AMBA_AXI_PROT
+    `undef AMBA_AXI_ID
+
+    `define AMBA_AXI_PROT
+    `AMBA_AXI_SLAVE_PORT(PCIE_M_AXI_LITE, unused, 32, 32)
```
Comment (abejgonzalez) [inline platforms/xilinx_alveo_u250/cl_firesim/design/overall_fpga_top.v:41]:
```suggestion
    `AMBA_AXI_WIRE(PCIE_M_AXI_LITE, unused, 32, 32)
```

**Comment by abejgonzalez** [comment]
TODO: Change to be based on `main` on #1671  is done.

**Comment by abejgonzalez** [comment]
@mergifyio rebase main

**Review by joonho3020 — APPROVED** [review]
Great stuff.
I would prefer defining all the macros in a single `fpga_top_macros.vh` file, but this seems fine as well.
